test/src/kbasesearchengine/test/main/SearchMethodsTest.java

```diff
@@ -669,4 +673,175 @@ private void compare(final SortingRule got, final SortingRule expected) {
+                        .withAdmin(true)
+                        .withPublic(true)))
+                .thenReturn(publicData);
+
+        when(is.searchTypes( kbasesearchengine.search.MatchFilter.getBuilder().build(),
+                new kbasesearchengine.search.AccessFilter()
+                        .withAccessGroups(set())
+                        .withPublic(true)))
+                .thenReturn(publicData);
+
+        //should not happen
+        when(is.searchTypes( kbasesearchengine.search.MatchFilter.getBuilder().build(),
+                new kbasesearchengine.search.AccessFilter()
+                        .withAccessGroups(set())
+                        .withPublic(false)))
+                .thenReturn(new HashMap<>());
+        return sm;
+    }
+    @Test
+    public void testAccessFilterAuthorizedUser() throws Exception{
+        SearchInterface sm = setUpTestAccessFilter();
+        //expected private results: SearchMethods changes integer results to longs
+        final Map<String, Long> privateData = new HashMap<>();
+        privateData.put("test", 1L);
+
+        final Map<String, Long> allData = new HashMap<>();
+        allData.put("test", 1L);
+        allData.put("notdefault", 1L);
+
+        final Map<String, Long> publicData = new HashMap<>();
+        publicData.put("notdefault", 1L);
+
+
+        final SearchTypesOutput res = sm.searchTypes(new SearchTypesInput()
+                        .withAccessFilter(new AccessFilter().withWithPrivate(1L).withWithPublic(1L))
+                        .withMatchFilter(new MatchFilter()),
+                "auser");
+
+        assertThat("should show both public and private", res.getTypeToCount(), is(allData));
+
+
+        final SearchTypesOutput res2 = sm.searchTypes(new SearchTypesInput()
+                        .withAccessFilter(new AccessFilter().withWithPrivate(0L).withWithPublic(1L))
+                        .withMatchFilter(new MatchFilter()),
+                "auser");
+        assertThat("should show only public", res2.getTypeToCount(), is(publicData));
+
+        final SearchTypesOutput res3 = sm.searchTypes(new SearchTypesInput()
+                        .withAccessFilter(new AccessFilter().withWithPrivate(1L).withWithPublic(0L))
+                        .withMatchFilter(new MatchFilter()),
+                "auser");
+        assertThat("should show only private", res3.getTypeToCount(), is(privateData));
+
+
+        final SearchTypesOutput res4 = sm.searchTypes(new SearchTypesInput()
+                        .withAccessFilter(new AccessFilter().withWithPrivate(0L).withWithPublic(0L))
+                        .withMatchFilter(new MatchFilter()),
+                "auser");
+        assertThat("should show only private", res4.getTypeToCount(), is(new HashMap<>()));
+
+    }
+
+    @Test
+    public void testAccessFilterUnauthorizedUser() throws Exception{
+        SearchInterface sm = setUpTestAccessFilter();
+        final Map<String, Long> publicData = new HashMap<>();
```

Comment:
For future reference, this is easier:

`final Map<String, Long> publicData = ImmutableMap.of("notdefault", 1L);`